src/problem.jl

```diff
@@ -0,0 +1,13 @@
+abstract type AbstractOptimizationProblem end
+
+struct OptimizationProblem{F,U,P,B,K} <: AbstractOptimizationProblem
+    f::F
+    u0::U
+    p::P
+    lb::B
+    ub::B
+    kwargs::K
+    function OptimizationProblem(f, p, u0=nothing; lb = nothing, ub = nothing, kwargs...)
```

Comment:
to keep with diffeq standards, `p` should be `p=DiffEqBase.NullParameters()`. Move `u0` to kwargs